pilot/pkg/serviceregistry/kube/controller/ambientindex_test.go

```diff
@@ -354,103 +354,6 @@ func TestAmbientIndex(t *testing.T) {
 		workloadapi.Protocol_HTTP)
 }
 
-func TestPodLifecycleWorkloadGates(t *testing.T) {
```

Comment:
While the underlying logic is changed, I would like to retain explicit unit tests verifying various combinations of lifecycle state changes produce the expected workload entry.

Reply:
Good call, added back